contracts/upgradeable_contracts/components/common/TokensRelayer.sol

```diff
@@ -25,7 +26,7 @@ abstract contract TokensRelayer is BasicAMBMediator, ReentrancyGuard {
         bytes calldata _data
     ) public returns (bool) {
         if (!lock()) {
-            bridgeSpecificActionsOnTokenTransfer(msg.sender, _from, chooseReceiver(_from, _data), _value);
+            bridgeSpecificActionsOnTokenTransfer(msg.sender, _from, chooseReceiver(_from, _data), _value, new bytes(0));
```

Comment:
Well, let's think more about this. Don't you think that a scenario when Alice would like to use an ERC677 token to make a transaction to call a method on a remote side after receiving the tokens?

E.g. she could call `token.transferAndCall(mediator, value, <recipient + abi encoded call>)`

So, the recipient on another chain will receive tokens by using `transferAndCall`, `onTokenTransfer` will be executed from the recipient contract and <abi encoded call> is passed as the third argument.

What do you think?

Reply:
Ok, that makes sense. How to distinguish between the following cases then:
* `token.transferAndCall(mediator, value, recipient)`
* `token.transferAndCall(mediator, value, abi.encodePacked(recipient, new bytes(0)))`

I think there are a lot of situations when data argument in the `onTokenTransfer` is empty. So, does it mean that we should always use `relayTokensAndCall` whenever `_data.length >= 20`?

Reply:
Yes, we will recommend to the apps' developers to add at least a byte to the alternative recipient address if they would like to use `relayTokensAndCall`. We will pass this byte to the onTokenTransfer method of the contract-recipient and it is up to them ignore it or use it for some purposes.